File: NPB-CPP-cadna/cadna_c-3.1.11/examplesFortran/ex7.f90
program ex7
  implicit none
  double precision :: r,r1,x,y,z
  integer :: i, nloop, ierr
  print *,'Example created on purpose to make CADNA fail'
  print *,'The same result r is computed for a number of iterations chosen by the user'
  print *,'The exact result is 1.4E-10'
  print *,'Enter the number of iterations'
  read *,nloop
  ierr = 0
  do i=1,nloop
    x=6.83561d+05
    y=6.83560d+05
    z=1.00000000007d0
    r = z - x
    r1 = z - y
    r = r + y
    r1 = r1 + x
    r1 = r1 - 2
    r = r + r1
    !      r = ((z-x)+y) + ((z-y)+x-2)
    if(r.ne.1.4d-10) then 
        ierr = ierr + 1
     endif
  enddo
  print *, 'Last value of r: ', r
  print *, 'Number of iterations when r is different from the exact result:', ierr 
end program ex7

File: NPB-CPP-cadna/cadna_c-3.1.11/examplesC/ex7.cc
#include "stdio.h"
#include <math.h>

int  main()
{
  double r=0,r1,x,y,z;
  int i, nloop, ierr;
  printf("Example created on purpose to make CADNA fail\n");
  printf("The same result r is computed for a number of iterations chosen by the user\n");
  printf("The exact result is 1.4E-10\n");
  printf("Enter the number of iterations: ");
  scanf("%d",&nloop);
  ierr = 0;
  for(i=0;i<nloop;i++){
    x=6.83561e+5;
    y=6.83560e+5;
    z=1.00000000007;
    r = z - x;
    r1 = z - y;
    r = r + y;
    r1 = r1 + x;
    r1 = r1 - 2;
    r = r + r1;
    //      r = ((z-x)+y) + ((z-y)+x-2)
    if(r != 1.4e-10) ierr = ierr + 1;
  }
  printf("Last value of r: %.15e\nNumber of iterations when r is different from the exact result: %d\n", r ,ierr);
  return 0;
}

